src/App.jsx

```diff
@@ -1,30 +1,52 @@
 import ReactDOM from "react-dom/client";
+import { useState } from "react";
 import { BrowserRouter, Routes, Route } from "react-router-dom";
 import AddPage from "./pages/AddPage/AddPage";
 import TodoList from "./pages/TodoList/TodoList";
 import ConfigurationPage from "./pages/ConfigurationPage/ConfigurationPage";
 import Header from "./components/Header/Header";
 import Navbar from "./components/Navbar/Navbar";
+import { StatusLengthContext } from "../StatusLengthContext";
+
 const App = () => {
+  const [allLength, setAllLength] = useState([]);
+  const [completedLength, setCompletedLength] = useState([]);
+  const [uncompletedLength, setUncompletedLength] = useState([]);
+
   return (
     <BrowserRouter>
-      <>
+      <StatusLengthContext.Provider
+        value={{
+          allLength,
+          setAllLength,
+          completedLength,
+          setCompletedLength,
+          uncompletedLength,
+          setUncompletedLength,
+        }}
```

Comment:
`allLength` - `completedLength` = `uncompletedLength`. So for the uncompleted one you can skip defining a new state.

Reply:
+ it's also possible to do this in one state. Something like:
```js
const [todosCount, setTodosCount] = useState({ all: 0, completed: 0 });
```
Then, when you want to change it:
```js
setTodosCount({ all: todosCount.all + 1, completed: todosCount.completed +1 })
```

Reply:
Done.